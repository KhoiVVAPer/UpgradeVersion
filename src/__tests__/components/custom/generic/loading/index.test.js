import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Loading from '../../../../../components/custom/generic/loading';

const setUp = (props) => {
  const component = shallow(<Loading {...props} />);
  return component;
};

describe('components > custom > elements > Loading', () => {
  let component;
  const props = {};
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders loading image', () => {
    const input = component.find('Image').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders configuring text image', () => {
    const input = component.find('Text').length;
    const output = 1;
    expect(input).toEqual(output);
  });
});
