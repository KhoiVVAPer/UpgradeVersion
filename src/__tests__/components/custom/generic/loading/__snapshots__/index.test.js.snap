// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > Loading Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock />
    <react-native-mock>
      <react-native-mock>
        Language configuration is setting up. Please wait.
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;
